jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/S3ExplorerNode.kt

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.s3
+
+import com.intellij.openapi.project.Project
+import icons.AwsIcons
+import com.intellij.ide.util.treeView.AbstractTreeNode
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.OpenFileDescriptor
+
+import software.amazon.awssdk.services.s3.S3Client
+import software.aws.toolkits.core.s3.regionForBucket
+import software.aws.toolkits.jetbrains.core.AwsClientManager
+import javax.swing.tree.DefaultMutableTreeNode
+import javax.swing.tree.DefaultTreeModel
+
+import software.aws.toolkits.jetbrains.core.credentials.ProjectAccountSettingsManager
+import software.aws.toolkits.jetbrains.core.explorer.AwsExplorerNode
+import software.aws.toolkits.jetbrains.core.explorer.AwsExplorerResourceNode
+import software.aws.toolkits.jetbrains.core.explorer.AwsExplorerServiceRootNode
+import software.aws.toolkits.jetbrains.core.explorer.AwsNodeAlwaysExpandable
+
+class S3ServiceNode(project: Project) : AwsExplorerServiceRootNode(project, "S3"),
```

Comment:
You probably want to rebase off master - this has changed slightly the `AwsExplorerService` enum has a `displayName` that can be used here.